### PR TITLE
Create generic executor that adds annotations to any other executor

### PR DESCRIPTION
**Pull request by hkaiser**
- flyby: removed obsolete #if defined(HPX_HAVE_CXX17_VARIABLE_TEMPLATES)
- flyby: constexpr variables can be inline
- flyby: modernize execution traits

### CODE REVIEW

**Comment by hkaiser** [comment]
retest lsu

File: libs/core/concepts/include/hpx/concepts/has_member_xxx.hpp [hunk 1]
```diff
@@ -12,16 +12,6 @@
...
 
-#if defined(HPX_HAVE_CXX17_VARIABLE_TEMPLATES)
-#define HPX_HAS_MEMBER_XXX_TRAIT_DEF_CXX17(MEMBER)                             \
-    template <typename T>                                                      \
-    constexpr bool HPX_PP_CAT(HPX_PP_CAT(has_, MEMBER), _v) =                  \
-        HPX_PP_CAT(has_, MEMBER)::value;                                       \
-    /**/
-#else
-#define HPX_HAS_MEMBER_XXX_TRAIT_DEF_CXX17(MEMBER) /**/
-#endif
```
Comment (msimberg) [inline libs/core/concepts/include/hpx/concepts/has_member_xxx.hpp:23]: Just curious about the history of this: were varable templates at some point not going to make it into C++14?

**Review by msimberg — APPROVED** [review]
Thanks @hkaiser!

Performance test CI failes due to #5405.

File: libs/core/concepts/include/hpx/concepts/has_member_xxx.hpp (hunk 1 above)
Comment (hkaiser) [reply libs/core/concepts/include/hpx/concepts/has_member_xxx.hpp:23]: That's exactly the question I asked myself :/ I must have been asleep while adding this option.

**Comment by hkaiser** [comment]
The failing tests on CirclCI are unrelated.

**Pull request merged**